recipes/cpython/all/conanfile.py

```diff
@@ -236,11 +253,145 @@ def generate(self):
         else:
             self._generate_autotools()
 
-    def _patch_sources(self):
-        apply_conandata_patches(self)
+    def _msvc_project_path(self, name):
+        return os.path.join(self.source_folder, "PCBuild", f"{name}.vcxproj")
+
+    def _regex_replace_in_file(self, filename, pattern, replacement):
+        content = load(self, filename)
+        content = re.sub(pattern, replacement, content)
+        save(self, filename, content)
+
+    def _inject_conan_props_file(self, project_basename, dep_name, condition=True):
+        if condition:
+            search = '<Import Project="python.props" />'
+            replace_in_file(self,
+                            self._msvc_project_path(project_basename),
+                            search,
+                            search + f'<Import Project="{self.generators_folder}/conan_{dep_name}.props" />')
+
+    def _patch_setup_py(self):
         setup_py = os.path.join(self.source_folder, "setup.py")
         if Version(self.version) < "3.10":
             replace_in_file(self, setup_py, ":libmpdec.so.2", "mpdec")
+
+        if self.options.get_safe("with_curses", False):
+            # FIXME: this will link to ALL libraries of ncurses. Only need to link to ncurses(w) (+ eventually tinfo)
+            ncurses_info = self.dependencies["ncurses"].cpp_info.aggregated_components()
+            replace_in_file(self, setup_py,
+                "curses_libs = ",
+                "curses_libs = {} #".format(repr(ncurses_info.libs + ncurses_info.system_libs)))
```

Comment:
```suggestion
            libcurses = self.dependencies["ncurses"].cpp_info.components["libcurses"]
            tinfo = self.dependencies["ncurses"].cpp_info.components["tinfo"]
            libs = libcurses.libs + libcurses.system_libs + info.libs + tinfo.system_libs
            replace_in_file(self, setup_py,
                "curses_libs = ",
                "curses_libs = {} #".format(repr(libs)))
```
Might as well include it since it's a simple fix.